### DISCUSSION TITLE
default installation scope

### DISCUSSION THREAD

**Discussion by dpprdan** [microsoft/winget-pkgs#76300] [Q&A] [answered]
I gather from [this discussion](https://github.com/microsoft/winget-pkgs/discussions/16219#discussioncomment-827014) that winget runs the installer with machine/admin scope if no scope is provided. Is that still correct (always?) and is that documented somewhere more "official"?

_When_ there are user and machine scopes in the manifest, then the [default is to install with user scope](https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/settings/settings.schema.0.2.json#L55-L63), correct? (The default isn't documented in the [settings docs](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#scope) BTW, other defaults (e.g. `telemetry`) and settings (e.g. `ignoreWarnings`, `disableInstallNotes`, ...) are missing as well).

**Comment by jedieaston** [comment, marked as answer]
I don't think it's documented, but if you're running in a non-elevated context, the default should be user scope if both are defined (based on some testing of `winget install vscode`). 

@denelon is this guaranteed anywhere? The client currently does this, but I'm not sure if it's guaranteed to do it forever.

**Reply by denelon** [reply]
The default is generally user scope if the user doesn't specify in arguments or settings, but it also depends on the installer itself. In some cases, if you are in an administrator prompt, the installer will essentially do what it's designed to do. We're looking to improve validation and looking at which registry keys are written as the primary hint. As we get more validation in place to test the cartesian of install configurations, we're able to enrich the manifest which improves the experience. Since we're working in this area, I'll look at what we can do in terms of documentation to help build some kind of logical flow to help users understand the default behaviors, what they can customize, and when that might not work.
